From patchwork Fri Jan 28 13:10:05 2022
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Michel Lespinasse <michel@lespinasse.org>
X-Patchwork-Id: 12728533
Return-Path: <owner-linux-mm@kvack.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from kanga.kvack.org (kanga.kvack.org [205.233.56.17])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 182EEC433FE
	for <linux-mm@archiver.kernel.org>; Fri, 28 Jan 2022 13:11:15 +0000 (UTC)
Received: by kanga.kvack.org (Postfix)
	id C5DE26B00A0; Fri, 28 Jan 2022 08:10:16 -0500 (EST)
Received: by kanga.kvack.org (Postfix, from userid 40)
	id 6C2896B00A1; Fri, 28 Jan 2022 08:10:16 -0500 (EST)
X-Delivered-To: int-list-linux-mm@kvack.org
Received: by kanga.kvack.org (Postfix, from userid 63042)
	id C0DC96B00A0; Fri, 28 Jan 2022 08:10:15 -0500 (EST)
X-Delivered-To: linux-mm@kvack.org
Received: from forelay.hostedemail.com (smtprelay0245.hostedemail.com
 [216.40.44.245])
	by kanga.kvack.org (Postfix) with ESMTP id 866546B0099
	for <linux-mm@kvack.org>; Fri, 28 Jan 2022 08:10:14 -0500 (EST)
Received: from smtpin11.hostedemail.com (10.5.19.251.rfc1918.com
 [10.5.19.251])
	by forelay01.hostedemail.com (Postfix) with ESMTP id 37DE5181C49C5
	for <linux-mm@kvack.org>; Fri, 28 Jan 2022 13:10:14 +0000 (UTC)
X-FDA: 79079729148.11.F9415B5
Received: from server.lespinasse.org (server.lespinasse.org [63.205.204.226])
	by imf31.hostedemail.com (Postfix) with ESMTP id C250E20011
	for <linux-mm@kvack.org>; Fri, 28 Jan 2022 13:10:12 +0000 (UTC)
DKIM-Signature: v=1; a=ed25519-sha256; c=relaxed/relaxed;
 d=lespinasse.org; i=@lespinasse.org; q=dns/txt; s=srv-52-ed;
 t=1643375407; h=from : to : cc : subject : date : message-id :
 in-reply-to : references : mime-version : content-transfer-encoding :
 from; bh=1qNQozrzr3u02685KSOX9mmqDOdBZq9NnzsUpi2IPaw=;
 b=6BYWY9OeTakFO1yedYXP/YDk5XWFtpTASfvFH9l3bWr14MWG0qakCH6lLPxP47W8w6cKt
 Kf50V6wlrfBBZOYAw==
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=lespinasse.org;
 i=@lespinasse.org; q=dns/txt; s=srv-52-rsa; t=1643375407; h=from : to
 : cc : subject : date : message-id : in-reply-to : references :
 mime-version : content-transfer-encoding : from;
 bh=1qNQozrzr3u02685KSOX9mmqDOdBZq9NnzsUpi2IPaw=;
 b=ueeE+B3WdhuSeZQhwoRdlPRcMIViOulAB79HXUYcoOLo++ZiI760j4qwPv9uDOQSTzzpW
 /SmFc0NJcAzQ9qS5VYlPofJcymNqAmvkn6oRVdcZSEiYya1FVJhCUjzwdTYfaaZ6pMonUkv
 0OOV3vWA6K4eYa/ht98YCHS0FBn6Vz4TjjnJq+hI//DdVqBC4orjnmU5VQ78FjKvdtwt6MH
 vi7Xyh7KIsSX2Hez+ibTnIMw2QdzCRnhJAZvjvOY7LBYpe/95Nwt4i0LpN9uv6NjtFtAhcv
 dWRPzuo3EhJ+uOmpw35SZ2O3c6niIb76XF8wQwAAjmAt9JkEuQjr5Fp+TQJg==
Received: from zeus.lespinasse.org (zeus.lespinasse.org [IPv6:fd00::150:0])
	by server.lespinasse.org (Postfix) with ESMTPS id 4A351160AEE;
	Fri, 28 Jan 2022 05:10:07 -0800 (PST)
Received: by zeus.lespinasse.org (Postfix, from userid 1000)
	id 2AF3520477; Fri, 28 Jan 2022 05:10:07 -0800 (PST)
From: Michel Lespinasse <michel@lespinasse.org>
To: Linux-MM <linux-mm@kvack.org>,
	linux-kernel@vger.kernel.org,
	Andrew Morton <akpm@linux-foundation.org>
Cc: kernel-team@fb.com,
	Laurent Dufour <ldufour@linux.ibm.com>,
	Jerome Glisse <jglisse@google.com>,
	Peter Zijlstra <peterz@infradead.org>,
	Michal Hocko <mhocko@suse.com>,
	Vlastimil Babka <vbabka@suse.cz>,
	Davidlohr Bueso <dave@stgolabs.net>,
	Matthew Wilcox <willy@infradead.org>,
	Liam Howlett <liam.howlett@oracle.com>,
	Rik van Riel <riel@surriel.com>,
	Paul McKenney <paulmck@kernel.org>,
	Song Liu <songliubraving@fb.com>,
	Suren Baghdasaryan <surenb@google.com>,
	Minchan Kim <minchan@google.com>,
	Joel Fernandes <joelaf@google.com>,
	David Rientjes <rientjes@google.com>,
	Axel Rasmussen <axelrasmussen@google.com>,
	Andy Lutomirski <luto@kernel.org>,
	Michel Lespinasse <michel@lespinasse.org>
Subject: [PATCH 34/35] [PATCH v2 34/35] powerpc/mm: define
 ARCH_SUPPORTS_SPECULATIVE_PAGE_FAULT
Date: Fri, 28 Jan 2022 05:10:05 -0800
Message-Id: <20220128131006.67712-35-michel@lespinasse.org>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20220128131006.67712-1-michel@lespinasse.org>
References: <20220128131006.67712-1-michel@lespinasse.org>
MIME-Version: 1.0
X-Rspamd-Server: rspam03
X-Rspamd-Queue-Id: C250E20011
X-Stat-Signature: a66qmjr95ctiwat4qdhuxuuer4p8kaix
Authentication-Results: imf31.hostedemail.com;
	dkim=pass header.d=lespinasse.org header.s=srv-52-ed header.b=6BYWY9Oe;
	dkim=pass header.d=lespinasse.org header.s=srv-52-rsa header.b=ueeE+B3W;
	dmarc=pass (policy=none) header.from=lespinasse.org;
	spf=pass (imf31.hostedemail.com: domain of walken@lespinasse.org designates
 63.205.204.226 as permitted sender) smtp.mailfrom=walken@lespinasse.org
X-Rspam-User: nil
X-HE-Tag: 1643375412-226236
X-Bogosity: Ham, tests=bogofilter, spamicity=0.000000, version=1.2.4
Sender: owner-linux-mm@kvack.org
Precedence: bulk
X-Loop: owner-majordomo@kvack.org
List-ID: <linux-mm.kvack.org>

Set ARCH_SUPPORTS_SPECULATIVE_PAGE_FAULT so that the speculative fault
handling code can be compiled on this architecture.

Signed-off-by: Michel Lespinasse <michel@lespinasse.org>
---
 arch/powerpc/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/powerpc/Kconfig b/arch/powerpc/Kconfig
index b779603978e1..5f82bc7eee0b 100644
--- a/arch/powerpc/Kconfig
+++ b/arch/powerpc/Kconfig
@@ -144,6 +144,7 @@ config PPC
 	select ARCH_STACKWALK
 	select ARCH_SUPPORTS_ATOMIC_RMW
 	select ARCH_SUPPORTS_DEBUG_PAGEALLOC	if PPC_BOOK3S || PPC_8xx || 40x
+	select ARCH_SUPPORTS_SPECULATIVE_PAGE_FAULT if PPC_BOOK3S_64
 	select ARCH_USE_BUILTIN_BSWAP
 	select ARCH_USE_CMPXCHG_LOCKREF		if PPC64
 	select ARCH_USE_MEMTEST

